watertap/unit_models/nanofiltration_0D.py

```diff
@@ -0,0 +1,675 @@
+            doc="""Indicates whether this model will be dynamic or not,
+    **default** = False. Product blocks are always steady-state.""",
+        ),
+    )
+    CONFIG.declare(
+        "has_holdup",
+        ConfigValue(
+            default=False,
+            domain=In([False]),
+            description="Holdup construction flag - must be False",
+            doc="""Product blocks do not contain holdup, thus this must be
+    False.""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package",
+        ConfigValue(
+            default=useDefault,
+            domain=is_physical_parameter_block,
+            description="Property package to use for mixer",
+            doc="""Property parameter object used to define property
+    calculations,
+    **default** - useDefault.
+    **Valid values:** {
+    **useDefault** - use default package from parent model or flowsheet,
+    **PropertyParameterObject** - a PropertyParameterBlock object.}""",
+        ),
+    )
+    CONFIG.declare(
+        "property_package_args",
+        ConfigDict(
+            implicit=True,
+            description="Arguments to use for constructing property packages",
+            doc="""A ConfigDict with arguments to be passed to a property
+    block(s) and used when constructing these,
+    **default** - None.
+    **Valid values:** {
+    see property package for documentation.}""",
+        ),
+    )
+    CONFIG.declare(
+        "include_pressure_balance",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Whether to include pressure balance for retentate side.",
+        ),
+    )
+    CONFIG.declare(
+        "has_retentate_pressure_drop",
+        ConfigValue(
+            default=False,
+            domain=Bool,
+            description="Whether to include pressure drop in the retentate side.",
+        ),
+    )
+    CONFIG.declare(
+        "include_temperature_equality",
+        ConfigValue(
+            default=True,
+            domain=Bool,
+            description="Whether to include temperature equality constraints",
+            doc="""Argument indicating whether temperature equality should be
+            inlcluded. If this is False, no energy balance constraint will be
+            written.""",
+        ),
+    )
+    CONFIG.declare(
+        "electroneutrality_ion",
+        ConfigValue(
+            default="Cl_-",
+            domain=str,
+            description="Balancing ion to use to maintain electroneutrality",
+            doc="""Name of ion to use to maintain electroneutrality in
+            permeate stream. If an ion is provided, the split fraction
+            for this ion will be solved implicitly to ensure electroneutrality
+            is maintained in the permeate stream. If None, user must provide
+            spl;it fractions for all monovalent ions and electroneutrality is
+            not guaranteed..""",
+        ),
+    )
```

Comment:
Actually, my response in that case is that you need to do data reconciliation to get an electroneutral solution, otherwise your data is not correct.

The model does support `None` for the electroneutrality species to allow you to forgo that constraint (and replaces it with a standard split fraction) for cases where the user wants to do that. As for doing a check on electroneutrality, to me that is a post-check the user should be doing, as there are too many edge cases that could occur.